### PR TITLE
Michal and Bosse

### PR DESCRIPTION
**Pull request by BoSott**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
 
 The Goal is to **practise writing readable, maintainable and reliable code collaboratively.**
```
Comment (hn437) [inline README.md]: Maybe you could rephrase it, saying this is a practice exams which was created in a seminar by students, but without telling the grading or rephrasing the exact task

File: README.md [hunk 2]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
...
 
-## Group Exercise
 
-1. One student of your group forks the code from [https://github.com/advanced-geoscripting-2021/random_walker.git](https://github.com/advanced-geoscripting-2021/random_walker.git)
+## Write an extended random walk program 
 
-2. This student invites the other student as a collaborator to the forked repository. Now you can both work on the code.
+In this repo you find a basic implementation of a [random walk simulation](https://en.wikipedia.org/wiki/Random_walk) in 2-dimensional space taken from [this blogpost](https://www.geeksforgeeks.org/random-walk-implementation-python/). Running the code yields an image which shows the path of the random walk.
```
Comment (hn437) [inline README.md:5]: Nice idea to link the repo and wiki!

File: README.md [hunk 3]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
...
-## Group Exercise
 
-1. One student of your group forks the code from [https://github.com/advanced-geoscripting-2021/random_walker.git](https://github.com/advanced-geoscripting-2021/random_walker.git)
+## Write an extended random walk program 
 
-2. This student invites the other student as a collaborator to the forked repository. Now you can both work on the code.
+In this repo you find a basic implementation of a [random walk simulation](https://en.wikipedia.org/wiki/Random_walk) in 2-dimensional space taken from [this blogpost](https://www.geeksforgeeks.org/random-walk-implementation-python/). Running the code yields an image which shows the path of the random walk. 
 
-3. Adapt the code to fulfil the requirements (see below).
+![random_walk](rand_walk_5_100.png)
```
Comment (hn437) [inline README.md:9]: Good idea to mark starting and endpoint! Maybe you could describe which symbol is the start/end, but looks good and is some useful additional info

File: README.md [hunk 4]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
...
-2. Add a "landscape" in which the random walkers are walking in which contains obstacles which the walkers cannot cross (e.g. a lake)
-3. Invent and implement another functionality of your own.
+  -sp, --start_point BOOLEAN    Specify whether the walkers shall start from
+                                the same point or not, Default is False
 
-Be creative here! :) 
+  --help                        Show this message and exit.
 
-## Code Review 
+````
```
Comment (hn437) [inline README.md]: Really neatly formatted! and really good solution, makes it easy to specify and understand which parameter you're setting

File: README.md [hunk 5]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
...
 
-Review the code of another group: (tuesday afternoon or wednesday morning)
+## Example
+``random_walker run --total_steps 10000 --step_size 2 -l False``
+
+4. Document the dependencies and instructions of how to run the program in your README.md.
```
Comment (hn437) [inline README.md]: you may delete this line...

File: README.md [hunk 6]
```diff
@@ -4,52 +4,68 @@ This is a **group exercise**, so you should be working in pairs of two students.
...
+1. clone repository
+2. setup a virtual environment
+3. activate the virtual environment
+4. install required libs and packages locally (see requirements.txt)
 
-5. Improve your code based on the review you got. 
+#### For the advanced geoscripting course:
+- activate the corresponding virtual environment
+- run ``conda install -c conda-forge click``
+- run ``pip install --editable .`` (is this actually necessary? can someone try to run it without this step?)
```
Comment (hn437) [inline README.md]: Yes, it is necessary!

File: README.md (hunk 2 above)
Comment (hn437) [inline README.md:5]: maybe you could add that the image will be shown to the user as a plot, but safed as a file as well

File: main.py [hunk 7]
```diff
@@ -4,37 +4,337 @@
...
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] - 1
+# user input – click
+_total_steps_option = [
+    click.option(
+        "--total_steps",
+        "-ts",
+        default=10000,
+        type=int,
+        help="Specify the number of total steps for the random walker, Default is 10,000",
```
Comment (hn437) [inline main.py]: nice idea to add the helping lines for each option

Comment (hn437) [inline main.py]: Is it per walker or combined for all walkers?

File: main.py [hunk 8]
```diff
@@ -4,37 +4,337 @@
...
+    click.option(
+        "--landscape",
+        "-l",
+        default=False,
+        type=bool,
+        help="Specify whether a grid landscape exists as base layer or not, Default is False",
+    )
+]
+
+_start_point_option = [
```
Comment (hn437) [inline main.py:55]: Good idea!

File: main.py [hunk 9]
```diff
@@ -4,37 +4,337 @@
...
-        x[i] = x[i - 1] - 1
-        y[i] = y[i - 1]
-    elif val == 3:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] + 1
-    else:
-        x[i] = x[i - 1]
-        y[i] = y[i - 1] - 1
+# user input – click
+_total_steps_option = [
```
Comment (hn437) [inline main.py:14]: with 1 step, the walker only stands (defining starting position) and doesn't step

File: main.py [hunk 10]
```diff
@@ -4,37 +4,337 @@
...
+    click.option(
+        "--total_steps",
+        "-ts",
+        default=10000,
+        type=int,
+        help="Specify the number of total steps for the random walker, Default is 10,000",
+    )
+]
+
+_total_walkers_option = [
```
Comment (hn437) [inline main.py:25]: if someone specifies 0 walker, the script is not exited but an empty plot is generated. However, the user get's the error that no handle was found to put in legend. Do you want it like this or should it be aborted?

File: README.md (hunk 6 above)
Comment (BoSott) [reply README.md]: thanks:)

File: main.py (hunk 7 above)
Comment (BoSott) [reply main.py]: this is for each walker

**Pull request closed without merging**